### DISCUSSION TITLE
Reservation of Iprange instead of only sending CIDR

### DISCUSSION THREAD

**Discussion by omerzubair** [Azure/ipam#224] [Ideas]
Hi @DCMattyG , 
the reservations work perfectly, when you issue a CIDR and it gives the next one. 
However, sometimes you want to reserve a iprange in the middle of a massive Space/Block, so that no project uses that iprange. 
Therefore just a question if we can reserve a complete IP range . 

I guess another way could be we can create an orphaned VNET sitting in subscription with Iprange we want it reserved. could that be the answer to this idea submitted?

**Comment by DCMattyG** [comment]
Hey there @omerzubair, there are two ways to accomplish this already within the Azure IPAM tool...

If you simply want to mark a large CIDR Block as utilized, perhaps outside the purview of the Azure IPAM tool (externally), there is a mechanism for this today called **External Networks**. Once an External Network is defined, the Azure IPAM tool will mark this space as consumed within the target Block, and will ignore this range when requesting new Reservations. You can find this feature here in the tool:

![image](https://github.com/Azure/ipam/assets/12955732/e20ff993-9ebc-4678-a5ce-8456f961f9a0)

![image](https://github.com/Azure/ipam/assets/12955732/cfe5949c-a88a-4b5d-bfd4-652c3019d604)

This can also be done via API as well:

![image](https://github.com/Azure/ipam/assets/12955732/d07299e3-df57-41b7-bea8-02b4785780a2)

The other option, if you're planning to use the CIDR space for a Virtual Network, is the ability to reserve a **specific** CIDR Block when calling the Reservations API:

![image](https://github.com/Azure/ipam/assets/12955732/5c12c86d-a58e-41b1-ba84-0c43cf4481c7)

This will check if the CIDR space is indeed available, then issue a Reservation for the desired Block which can be used to create a Virtual Network in Azure.

I hope one of those two mechanisms makes sense for your use case. Please let me know if you have any further questions!

**Reply by omerzubair** [reply]
Yup both will suffice I personally like the reservations option i.e you can simply check if the iprange is avail or not.
if yes then it will issue it otherwise it will throw error {"error":"Requested CIDR overlaps existing network(s)."}
